task_2/src/controllers/userController.ts

```diff
@@ -1,246 +1,111 @@
 
-    // check if all the required fields are provided
-    const { error, value }: { error: ValidationError; value: UserRequestBody } = userRequestSchema.validate(req.body)
-
-    if (error) {
+    if (schemaValidation.error) {
       return res.status(400).json({
         success: false,
-        message: error.details[0].message,
-        error,
+        message: schemaValidation.error.details[0].message,
+        error: schemaValidation.error,
       })
     }
 
-    const users = await _getUsers()
-    const isLoginNotAvailable = users.find((user) => user.login === value.login)
-
-    if (isLoginNotAvailable) {
-      return res.status(400).json({
-        success: false,
-        message: LOGIN_NOT_AVAILABLE_MESSAGE,
-        data: value,
-      })
+    const userId = parseInt(req.params.id)
+    const updatedUser: IUser = {
+      id: userId,
+      login: schemaValidation.value.login,
+      password: schemaValidation.value.password,
+      age: schemaValidation.value.age,
+      is_deleted: false,
     }
 
-    const updatedUser: User = {
-      id: user.id,
-      login: value.login,
-      password: value.password,
-      age: value.age,
-      isDeleted: user.isDeleted,
-    }
+    const result = await userService.update(updatedUser)
 
-    const updatedUserIndex = users.findIndex((user) => user.id === updatedUser.id)
-    users[updatedUserIndex] = updatedUser
-
-    await fs.writeFile(databaseFilePath, jsonStringfy(users))
+    if (result?.error) {
+      return res.status(400).json(result.error)
+    }
 
-    res.status(200).json({
-      success: true,
-      message: USER_UPDATED_MESSAGE,
-      data: updatedUser,
-    })
+    res.status(200).json(result.response)
   } catch (error) {
-    res.status(500).json({
-      success: false,
-      message: SOMETHING_WENT_WRONG_MESSAGE,
-      error: error.message,
-    })
+    res.status(500).json(getGenericErrorMessage(error))
   }
 }
 
 export const deleteUser = async (req: Request, res: Response) => {
   try {
-    // check if the user with the provided ID exists
-    const userId = req.params.id
-    const user = await _getUserById(userId)
+    const userId = parseInt(req.params.id)
+    const result = await userService.delete(userId)
 
-    if (!user) {
-      return res.status(400).json({
-        success: false,
-        error: NO_USER_FOUND_MESSAGE,
-        data: userId,
-      })
+    if (result?.error) {
+      return res.status(400).json(result.error)
     }
 
-    const users = await _getUsers()
-    const deletedUser: User = { ...user, isDeleted: true }
-
-    const deletedUserIndex = users.findIndex((user) => user.id === deletedUser.id)
-    users[deletedUserIndex] = deletedUser
-
-    await fs.writeFile(databaseFilePath, jsonStringfy(users))
-
-    res.status(200).json({
-      success: true,
-      message: USER_DELETED_MESSAGE,
-      data: deletedUser,
-    })
+    res.status(200).json(result.response)
```

Comment:
in delete endpoints response status won't be 200 and no message will be sent!

(from google)
If a DELETE method is successfully applied, there are several response status codes possible:
1. A 202 ( Accepted ) status code if the action will likely succeed but has not yet been enacted.
2. A 204 ( No Content ) status code if the action has been enacted and no further information is to be supplied.

Reply:
Updated accordingly